server/database/index.js

```diff
@@ -17,75 +18,80 @@ const ImageSchema = mongoose.Schema({
+    Image.find({ '_id': productId })
+      .select('large')
+      .then(((images) => cb(null, images[0].large)))
+      .catch((err) => {
+        console.log('GETTING LARGE IMAGES ERROR = ', err);
+        cb(err, null);
+      });
+  },
 
-var getColors = (productId) => {
-  Image.find({ '_id': productId })
-    .select('colors')
-    .then(((images) => {
-      console.log('images = ', images);
-    }))
-    .catch((err) => {
-      console.log('GETTING COLOR THUMBNAILS IMAGES ERROR = ', err);
-    });
-};
+  getRegular: (productId, cb) => {
+    Image.find({ '_id': productId })
+      .select('regular')
+      .then(((images) => cb(null, images[0].regular)))
+      .catch((err) => {
+        console.log('GETTING REGULAR IMAGES ERROR = ', err);
+        cb(err, null);
+      });
+  },
 
+  getColors: (productId, cb) => {
+    Image.find({ '_id': productId })
+      .select('colors')
+      .then(((images) => cb(null, images[0].colors)))
+      .catch((err) => {
+        console.log('GETTING COLOR THUMBNAILS IMAGES ERROR = ', err);
+        cb(err, null);
+      });
+  },
 
-var getSizeService = (productId) => {
-  Image.find({ '_id': productId })
-    .select('sizeService')
-    .then(((images) => {
-      console.log('images = ', images);
-    }))
-    .catch((err) => {
-      console.log('GETTING HAND-DRAWN IMAGES ERROR = ', err);
-    });
-};
-
-var getLargeThumbnails = (productId) => {
-  Image.find({ '_id': productId })
-    .select('largeThumbnails')
-    .then(((images) => {
-      console.log('images = ', images);
-    }))
-    .catch((err) => {
-      console.log('GETTING LARGE THUMBNAILS IMAGES ERROR = ', err);
-    });
-};
+  getSizeService: (productId, cb) => {
+    Image.find({ '_id': productId })
+      .select('sizeService')
+      .then(((images) => cb(null, images[0].sizeService)))
+      .catch((err) => {
+        console.log('GETTING SIZE-SERVICE IMAGES ERROR = ', err);
+        cb(err, null);
+      });
+  },
 
+  getLargeThumbnails: (productId, cb) => {
+    Image.find({ '_id': productId })
+      .select('largeThumbnails')
+      .then(((images) => cb(null, images[0].largeThumbnails)))
+      .catch((err) => {
+        console.log('GETTING LARGE THUMBNAILS IMAGES ERROR = ', err);
+        cb(err, null);
+      });
+  },
 
-var getMediumThumbnails = (productId) => {
-  Image.find({ '_id': productId })
-    .select('mediumThumbnails')
-    .then(((images) => {
-      console.log('images = ', images);
-    }))
-    .catch((err) => {
-      console.log('GETTING MEDIUM THUMBNAILS IMAGES ERROR = ', err);
-    });
+  getMediumThumbnails: (productId, cb) => {
+    Image.find({ '_id': productId })
+      .select('mediumThumbnails')
+      .then(((images) => cb(null, images[0].mediumThumbnails)))
+      .catch((err) => {
+        console.log('GETTING MEDIUM THUMBNAILS IMAGES ERROR = ', err);
+        cb(err, null);
+      });
+  }
 };
 
+Object.keys(db).map((method) => db[method] = Promise.promisify(db[method]));
```

Comment:
This is a good idea, I should try this. Promisifying prior to exporting the functions.